base/channels.jl

```diff
@@ -619,6 +619,7 @@ function iterate(c::Channel, state=nothing)
             end
         end
     else
+        check_channel_state(c)
```

Comment:
You could inline this implementation, or perhaps more easily, just delete the `isopen` check at the top and let `take!` handle everything (as it already implements everything necessary)